File: q1161/q1161.cpp
#include <queue>
using namespace std;

class Solution
{
public:
    int maxLevelSum(TreeNode *root)
    {

        int currMaxSum = root->val;
        int currLevel = 1, maxSumLevel = 1;

        // Initialize the queue for BFS
        queue<TreeNode *> q;
        q.push(root);

        while (!q.empty())
        {
            int num = q.size();
            int levelSum = 0;
            for (int i = 0; i < num; i++)
            {
                TreeNode *node = q.front();
                levelSum += node->val;
                if (node->left != nullptr)
                    q.push(node->left);
                if (node->right != nullptr)
                    q.push(node->right);
                q.pop();
            }
            if (levelSum > currMaxSum)
            {
                maxSumLevel = currLevel;
                currMaxSum = levelSum;
            }
            currLevel++;
        }
        return maxSumLevel;
    }
};

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

File: q1161/q1161.py
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        max_sum = root.val
        max_level = 1
        curr_level = 1

        queue = [root]
        while queue:
            count = len(queue)
            level_sum = 0
            for _ in range(count):
                curr_node = queue.pop(0)
                level_sum += curr_node.val
                if curr_node.left: queue.append(curr_node.left)
                if curr_node.right: queue.append(curr_node.right) 
            if level_sum > max_sum:
                max_sum = level_sum
                max_level = curr_level
            curr_level += 1
        return max_level
